src/main/scala/com/google/cloud/genomics/spark/examples/GenomicsConf.scala

```diff
@@ -58,13 +58,18 @@ class GenomicsConf(arguments: Seq[String]) extends ScallopConf(arguments) {
   }
 
   def getReferences = {
-    this.references().split(",").map(contig => {
-      val data = contig.split(":")
-      (data(0), (data(1).toLong, data(2).toLong))
-    }).toMap
+    Contig.parseContigsFromCommandLine(this.references())
   }
 }
 
 class PcaConf(arguments: Seq[String]) extends GenomicsConf(arguments) {
   val numPc = opt[Int](default = Some(2))
+  val allReferences = opt[Boolean]()
```

Comment:
Happy New Year Nicole and Cassie!  Based on how [Contig](https://github.com/googlegenomics/utils-java/blob/master/src/main/java/com/google/cloud/genomics/utils/Contig.java) is currently implemented, it  probably would have to be done in parallel as a task per each chromosome.  The start and end positions would need to be provided, but that can easily be retrieved via [Coverage API](https://cloud.google.com/genomics/v1beta2/reference/readgroupsets/coveragebuckets/list) with no settings, as @cassiedoll mentioned in another post.

Even if Dataflow is used, it looks based on [lines 45-47 of GenomicsDatasetOptions](https://github.com/googlegenomics/dataflow-java/blob/b2554b8e4f62346a62f5310461f4d697f28f750a/src/main/java/com/google/cloud/genomics/dataflow/utils/GenomicsDatasetOptions.java#L45), that the boolean setting (`isAllContigs`) still works via contigs that [Contig]
(https://github.com/googlegenomics/utils-java/blob/master/src/main/java/com/google/cloud/genomics/utils/Contig.java) provides.  The PCA function based on [PCoAnalysis.java](https://github.com/googlegenomics/dataflow-java/blob/b2554b8e4f62346a62f5310461f4d697f28f750a/src/main/java/com/google/cloud/genomics/dataflow/functions/PCoAnalysis.java), looks like just a function that would be applied to the `PCollection` as shown in a similar example in [VariantSimilarity.java on lines 53-56](https://github.com/googlegenomics/dataflow-java/blob/b2554b8e4f62346a62f5310461f4d697f28f750a/src/main/java/com/google/cloud/genomics/dataflow/pipelines/VariantSimilarity.java#L53)

~p